kolibri/core/assets/src/core-actions.js

```diff
@@ -83,6 +83,78 @@ function _sessionState(data) {
   return state;
 }
 
+const masteryLoggingMap = {
+  id: 'pk',
+  start_timestamp: 'start_timestamp',
+  completion_timestamp: 'completion_timestamp',
+  end_timestamp: 'end_timestamp',
+  mastery_criterion: 'mastery_criterion',
+  complete: 'complete',
+  responsehistory: 'responsehistory',
+};
+
+function _masteryLoggingState(data) {
+  const state = {};
+  Object.keys(masteryLoggingMap).forEach((key) => {
+    state[key] = data[masteryLoggingMap[key]];
+  });
+  return state;
+}
+
+function _masteryLogModel(store) {
+  const mapping = {};
+  const masteryLog = store.state.core.logging.mastery;
+  Object.keys(masteryLoggingMap).forEach((key) => {
+    mapping[masteryLoggingMap[key]] = masteryLog[key];
+  });
+  mapping.summarylog = store.state.core.logging.summary.id;
+  return mapping;
+}
+
+const attemptLoggingMap = {
+  id: 'pk',
+  item: 'item',
+  start_timestamp: 'start_timestamp',
+  completion_timestamp: 'completion_timestamp',
+  end_timestamp: 'end_timestamp',
+  time_spent: 'time_spent',
+  complete: 'complete',
+  correct: 'correct',
+  answer: 'answer',
+  simple_answer: 'simple_answer',
+  interaction_history: 'interaction_history',
+};
+
+const attemptLoggingJSONifyKeys = {
+  answer: true,
+  interaction_history: true,
+};
```

Comment:
It looks like some values are stored as serialized JSON. (I now see this is the case with `extra_fields` in the content session logging also). This seems like an implementation detail of the database that should be hidden from the client. Would it be possible to move this kind of logic to the server API serialize/deserialize layer, and allow the JS client to treat all API content as regular JSON?